lib/bitcoin/BitcoinClient.ts

```diff
@@ -42,12 +41,21 @@ export default class BitcoinClient {
     private bitcoinPeerUri: string,
     bitcoinRpcUsername: string | undefined,
     bitcoinRpcPassword: string | undefined,
-    bitcoinWalletImportString: string,
+    bitcoinWalletImportString: string | undefined,
+    bitcoinWallet: IBitcoinWallet | undefined,
     private requestTimeout: number,
     private requestMaxRetries: number,
     private sidetreeTransactionFeeMarkupPercentage: number) {
 
-    this.bitcoinWallet = new BitcoinWallet(bitcoinWalletImportString);
+    if (bitcoinWalletImportString && bitcoinWallet) {
+      throw new SidetreeError(ErrorCode.BitcoinProcessorWalletAndImportStringSpecified, 'Both bitcoin wallet and import string are not allowed.');

```

Comment:
Nit: This error message is confusing to read for me. I think it will be clearer if we say something like `can only have either bitcoin wallet or import string configured, having both at the in the config is not allowed.` The message you used sounded like both are not allowed like "Both Ali and Isaac are not allowed to be here".

Reply:
Good point. Will update.

Reply:
Error no longer needed after refactor.